sympy/assumptions/handlers/sets.py

```diff
@@ -31,682 +33,650 @@ def _number(expr, assumptions):
         except TypeError:
             return False
 
-    @staticmethod
-    def Add(expr, assumptions):
-        """
-        Integer + Integer       -> Integer
-        Integer + !Integer      -> !Integer
-        !Integer + !Integer -> ?
-        """
-        if expr.is_number:
-            return AskIntegerHandler._number(expr, assumptions)
-        return test_closed_group(expr, assumptions, Q.integer)
-
-    @staticmethod
-    def Mul(expr, assumptions):
-        """
-        Integer*Integer      -> Integer
-        Integer*Irrational   -> !Integer
-        Odd/Even             -> !Integer
-        Integer*Rational     -> ?
-        """
-        if expr.is_number:
-            return AskIntegerHandler._number(expr, assumptions)
-        _output = True
-        for arg in expr.args:
-            if not ask(Q.integer(arg), assumptions):
-                if arg.is_Rational:
-                    if arg.q == 2:
-                        return ask(Q.even(2*expr), assumptions)
-                    if ~(arg.q & 1):
-                        return None
-                elif ask(Q.irrational(arg), assumptions):
-                    if _output:
-                        _output = False
-                    else:
-                        return
+for cls in (int, Integer):
+    IntegerPredicate.register(cls)(lambda expr, assumptions: True)
+
+for cls in (Exp1, GoldenRatio, ImaginaryUnit, Infinity, NaN, NegativeInfinity,
+            Pi, Rational, TribonacciConstant):
+    IntegerPredicate.register(cls)(lambda expr, assumptions: False)
```

Comment:
Can these sorts of examples not use `register_many`?

Reply:
Oh, right. I overlook them...